From 12d4c67669054f4ace3fd8715f4a29073861f4a5 Mon Sep 17 00:00:00 2001
From: Daniel Vetter <daniel.vetter@ffwll.ch>
Date: Sun, 4 Oct 2009 15:00:36 +0200
Patch-mainline: 2.6.33
References: fate#310916
Git-commit: 5c5a4359fe392b52b444134877fc4002be542b42
Subject: [PATCH 0042/2588] drm/i915: overlay: kill one more unnecessary
 uninterruptible sleep

I've simply overlooked one case in the conversion to interruptible
sleeps. Rectify this.

Also delete a leftover debug printk.

Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Signed-off-by: Eric Anholt <eric@anholt.net>
(cherry picked from commit 5c5a4359fe392b52b444134877fc4002be542b42)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/intel_overlay.c |   17 +++++++----------
 1 files changed, 7 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_overlay.c b/drivers/gpu/drm/i915/intel_overlay.c
index 972d715..f1bf0b0 100644
--- a/drivers/gpu/drm/i915/intel_overlay.c
+++ b/drivers/gpu/drm/i915/intel_overlay.c
@@ -286,16 +286,15 @@ static int intel_overlay_wait_flip(struct intel_overlay *overlay)
 	RING_LOCALS;
 
 	if (overlay->last_flip_req != 0) {
-		ret = i915_do_wait_request(dev, overlay->last_flip_req, 0);
-		if (ret != 0)
-			return ret;
-
-		overlay->last_flip_req = 0;
+		ret = i915_do_wait_request(dev, overlay->last_flip_req, 1);
+		if (ret == 0) {
+			overlay->last_flip_req = 0;
 
-		tmp = I915_READ(ISR);
+			tmp = I915_READ(ISR);
 
-		if (!(tmp & I915_OVERLAY_PLANE_FLIP_PENDING_INTERRUPT))
-			return 0;
+			if (!(tmp & I915_OVERLAY_PLANE_FLIP_PENDING_INTERRUPT))
+				return 0;
+		}
 	}
 
 	/* synchronous slowpath */
@@ -439,8 +438,6 @@ int intel_overlay_recover_from_interrupt(struct intel_overlay *overlay,
 				return ret;
 
 		case SWITCH_OFF_STAGE_2:
-			printk("switch off 2\n");
-
 			BUG_ON(!overlay->vid_bo);
 			obj = overlay->vid_bo->obj;
 
-- 
1.7.6

